src/sources/fluent/mod.rs

```diff
@@ -925,6 +1029,90 @@ mod tests {
         req.serialize(&mut Serializer::new(&mut buf)).unwrap();
         buf
     }
+
+    #[test]
+    fn output_schema_definition_vector_namespace() {
+        let config = FluentConfig {
+            address: SocketListenAddr::SocketAddr("0.0.0.0:24224".parse().unwrap()),
+            tls: None,
+            keepalive: None,
+            receive_buffer_bytes: None,
+            acknowledgements: false.into(),
+            connection_limit: None,
+            log_namespace: None,
+        };
+
+        let definition = config.outputs(LogNamespace::Vector)[0]
+            .clone()
+            .log_schema_definition
+            .unwrap();
+
+        let expected_definition =
+            Definition::new_with_default_metadata(Kind::bytes(), [LogNamespace::Vector])
+                .with_meaning(LookupBuf::root(), "message")
+                .with_metadata_field(&owned_value_path!("vector", "source_type"), Kind::bytes())
+                .with_metadata_field(
+                    &owned_value_path!("fluent", "tag"),
```

Comment:
I'd expect this to be failing since `tag` was inserted with

```rust
        log_namespace.insert_vector_metadata(&mut log, path!("tag"), path!("tag"), tag);

```

Reply:
I believe the test case didn't fail because this test is just comparing the schema definition to the expected, not actual events

Reply:
Which I guess begs the question, do we need to have _both_ types of unit tests and not just one or the other

Reply:
The helper @fuchsnj added can do a type comparison when you have events, so you could combine the two. I think we want to make sure, especially for `Legacy` that the shape of the events aren't changing.